engine/resource/src/resource_archive.cpp

```diff
@@ -45,7 +45,20 @@ namespace dmResourceArchive
 
     int              g_NumArchiveLoaders = 0;
     ArchiveLoader    g_ArchiveLoader[4];
-    FDecryptResource g_ResourceDecryption = dmResourceArchive::DecryptWithXtea;
+
+
+    Result DecryptWithXtea(void* buffer, uint32_t buffer_len)
```

Comment:
We recommend keeping private functions marked as "static", both for communicating to the developer, but also to not export the function outside of the translation unit.